scopes/code/hooks/use-code-params/use-code-params.ts

```diff
@@ -1,11 +1,17 @@
 import { useParams } from 'react-router-dom';
+import { useQuery } from '@teambit/ui.react-router.use-query';
 
 export type CodeRouteParams = {
   componentId: string;
   file: string | undefined;
+  version?: string;
 };
 
 export function useCodeParams(): CodeRouteParams {
+  // TODO - not sure this is the best way to do this. we might want to handle the query params in react-router aspect
+  // I think we should have a hook that keeps all qury parms and just overrides a specific one
```

Comment:
yes, I don't like it either.
I'd rather invert this pattern to get version from props.

We could make it use query params by default, but it will just couple the hook to a browser url, I don't like that either.

Reply:
so how do you suggest we proceed here? or do you want to leave it for later?